core/src/main/java/com/amplitude/core/utilities/JSONUtil.kt

```diff
@@ -157,50 +157,50 @@ internal fun JSONArray.toIntArray(): IntArray {
-    event.carrier = this.optString("carrier", null)
-    event.country = this.optString("country", null)
-    event.region = this.optString("region", null)
-    event.city = this.optString("city", null)
-    event.dma = this.optString("dma", null)
-    event.language = this.optString("language", null)
+    event.eventProperties = this.optionalJSONObject("event_properties", null)
+    event.userProperties = this.optionalJSONObject("user_properties", null)
+    event.groups = this.optionalJSONObject("groups", null)
+    event.appVersion = this.optionalString("app_version", null)
+    event.platform = this.optionalString("platform", null)
+    event.osName = this.optionalString("os_name", null)
+    event.osVersion = this.optionalString("os_version", null)
+    event.deviceBrand = this.optionalString("device_brand", null)
+    event.deviceManufacturer = this.optionalString("device_manufacturer", null)
+    event.deviceModel = this.optionalString("device_model", null)
+    event.carrier = this.optionalString("carrier", null)
+    event.country = this.optionalString("country", null)
+    event.region = this.optionalString("region", null)
+    event.city = this.optionalString("city", null)
+    event.dma = this.optionalString("dma", null)
+    event.language = this.optionalString("language", null)
     event.price = if (this.has("price")) this.getDouble("price") else null
     event.quantity = if (this.has("quantity")) this.getInt("quantity") else null
     event.revenue = if (this.has("revenue")) this.getDouble("revenue") else null
-    event.productId = this.optString("productId", null)
-    event.revenueType = this.optString("revenueType", null)
+    event.productId = this.optionalString("productId", null)
+    event.revenueType = this.optionalString("revenueType", null)
     event.locationLat = if (this.has("location_lat")) this.getDouble("location_lat") else null
     event.locationLng = if (this.has("location_lng")) this.getDouble("location_lng") else null
-    event.ip = this.optString("ip", null)
-    event.idfa = this.optString("idfa", null)
-    event.idfv = this.optString("idfv", null)
-    event.adid = this.optString("adid", null)
-    event.androidId = this.optString("android_id", null)
+    event.ip = this.optionalString("ip", null)
+    event.idfa = this.optionalString("idfa", null)
+    event.idfv = this.optionalString("idfv", null)
+    event.adid = this.optionalString("adid", null)
+    event.androidId = this.optionalString("android_id", null)
     event.eventId = if (this.has("event_id")) this.getInt("event_id") else null
     event.sessionId = this.getLong("session_id")
-    event.insertId = this.optString("insert_id", null)
+    event.insertId = this.optionalString("insert_id", null)
     event.library = if (this.has("library")) this.getString("library") else null
-    event.partnerId = this.optString("partner_id", null)
+    event.partnerId = this.optionalString("partner_id", null)
```

Comment:
do we need app set id too this time?

Reply:
Good catch, missed that when just looking at the http api docs.. We should add that into the api doc